src/shared/tpm2-util.c

```diff
@@ -1558,33 +1557,22 @@ int tpm2_seal(const char *device,
 
         log_debug("Marshalling private and public part of HMAC key.");
 
-        k = ALIGN8(sizeof(*private)) + ALIGN8(sizeof(*public)); /* Some roughly sensible start value */
-        for (;;) {
-                _cleanup_free_ void *buf = NULL;
-                size_t offset = 0;
-
-                buf = malloc(k);
-                if (!buf)
-                        return log_oom();
+        _cleanup_free_ void *blob = NULL;
+        size_t max_size = sizeof(*private) + sizeof(*public), blob_size = 0;
```

Comment:
can you please add in a comment why this buffer size will be sufficient? i.e. a short form of what you wrote in the commit msg?

Reply:
I have another PR (#26353) that does more work around marshalling, I can add a comment there clarifying the size requirement for marshalling/unmarshalling